### DISCUSSION TITLE
Principal derived from a Canister Signature public “key”, How to understand it?

### DISCUSSION THREAD

**Discussion by flyq** [nomeata/capture-the-ic-token#1] [Q&A]
![image](https://user-images.githubusercontent.com/16486808/129675886-607168ac-f932-423a-84ea-886bd346d8a3.png)
![image](https://user-images.githubusercontent.com/16486808/129675993-8e84bc5e-6322-4d1f-866d-e06eb8123c88.png)
![image](https://user-images.githubusercontent.com/16486808/129676480-383359ab-a6f3-4241-81b6-910c77b7e330.png)

To be honest, I am still quite confused about some of the techniques in this challenge.

Where is the Public key of Principal `qhngm-45l5d-nm6db-whcbh-dkq5h-n66gd-scpbe-tl3k7-dp3n7-t3liu-wae` generated from? Is it not genenrate from a Private key? I can't understand a Public key not generated from a private key.

>  derived from a Canister Signature public “key”

Where is the private key of Canister Signature public "key", the Canister Signature can be vertified by other platforms? like Ethereum or something can vertify the ECDSA?

**Comment by nomeata** [comment]
The principal `qhngm-45l5d-nm6db-whcbh-dkq5h-n66gd-scpbe-tl3k7-dp3n7-t3liu-wae` is in a way “owned” by the canister with canister id `6b4pv-sqaaa-aaaah-qaava-cai`. The canister (and only the canister) can “sign” requests with that principal, using the “[Canister Signature](https://sdk.dfinity.org/docs/interface-spec/index.html#canister-signatures)” scheme. There is no private key, because the canister doesn’t create the signature itself, but – in a manner of speaking – instructs the _subnet_, which has a secret key, to create the signature. But in a way that is still tied to that canister id.

**Reply by flyq** [reply]
canister's signature can be vertified by Ethereum?

**Reply by nomeata** [reply]
No, it’s IC-specific. I heard rumors that a dedicated Ethereum bridge is in the works, though.
